src/Npgsql/NpgsqlTypes/NpgsqlTimetz.cs

```diff
@@ -0,0 +1,287 @@
+        /// </summary>
+        /// <param name="hour">The hour (0 through 24).</param>
+        /// <param name="minute">The minute (0 through 60).</param>
+        /// <param name="second">The second (0 through 60).</param>
+        /// <param name="millisecond">The millisecond (0 through 1000).</param>
+        /// <param name="microsecond">The microsecond (0 through 1000).</param>
+        /// <param name="timeZone">A time zone offset from Coordinated Universal Time (UTC).</param>
+        public NpgsqlTimetz(int hour, int minute, int second, int millisecond, int microsecond, NpgsqlTimeZone timeZone)
+            : this(new NpgsqlTime(hour, minute, second, millisecond, microsecond), timeZone) { }
+
+        /// <summary>Represents the largest possible value of <see cref="NpgsqlTimetz"/>.</summary>
+        /// <value>The largest possible value of <see cref="NpgsqlTimetz"/>.</value>
+        public static NpgsqlTimetz MaxValue => new NpgsqlTimetz(NpgsqlTime.MaxValue, NpgsqlTimeZone.MinValue);
+
+        /// <summary>Represents the smallest possible value of <see cref="NpgsqlTimetz"/>.</summary>
+        /// <value>The smallest possible value of <see cref="NpgsqlTimetz"/>.</value>
+        public static NpgsqlTimetz MinValue => new NpgsqlTimetz(NpgsqlTime.MinValue, NpgsqlTimeZone.MaxValue);
+
+        /// <summary>Gets the time component of the timestamp represented by this instance.</summary>
+        /// <value>The time component.</value>
+        public NpgsqlTime Time => _time;
+
+        /// <summary>Gets the time zone component of the timestamp represented by this instance.</summary>
+        /// <value>The time zone component.</value>
+        public NpgsqlTimeZone TimeZone => _timeZone;
+
+        /// <summary>Gets the hour component of the time represented by this instance.</summary>
+        /// <value>The hour component.</value>
+        public int Hour => _time.Hour;
+
+        /// <summary>Gets the minute component of the time represented by this instance.</summary>
+        /// <value>The minute component.</value>
+        public int Minute => _time.Minute;
+
+        /// <summary>Gets the second component of the time represented by this instance.</summary>
+        /// <value>The second component.</value>
+        public int Second => _time.Second;
+
+        /// <summary>Gets the millisecond component of the time represented by this instance.</summary>
+        /// <value>The millisecond component.</value>
+        public int Millisecond => _time.Millisecond;
+
+        /// <summary>Gets the microsecond component of the time represented by this instance.</summary>
+        /// <value>The microsecond component.</value>
+        public int Microsecond => _time.Microsecond;
+
+        /// <summary>Gets the number of microseconds since the midnight.</summary>
+        /// <value>The number of microseconds since the midnight.</value>
+        public long Microseconds => _time.Microseconds;
```

Comment:
This is begging to get a typo somewhere :D

Reply:
What do you mean here? I'm not following.

Reply:
Yeah, I should have been clearer. We have 2 different properties here: `Microseconds` and `Microsecond`, where one returns `long` and another `int`. Getting a typo here it pretty easy. Maybe we should rename it?